### PR TITLE
ECOM-7262 Expose course run seats on program endpoint

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
@edx/ecommerce

### CODE REVIEW

**Comment by davec-edx** [comment]
Looks ok to me, my only question was why change from 'seats' to 'staff' in the test, but that got answered.

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -383,6 +383,7 @@ class Meta:
 class MinimalCourseRunSerializer(TimestampModelSerializer):
     image = ImageField(read_only=True, source='card_image_url')
     marketing_url = serializers.SerializerMethodField()
+    seats = SeatSerializer(many=True)
```
Comment (clintonb) [inline course_discovery/apps/api/serializers.py:386]: Why are we adding items to the minimal serializer? That goes against the minimal idea.

File: course_discovery/apps/api/tests/test_serializers.py [hunk 2]
```diff
@@ -871,26 +872,26 @@ def test_use_full_course_serializer(self):
         request = make_request()
         program = self.create_program()
 
-        # Assert that the seats are not available when fetched without any flag
```
Comment (clintonb) [inline course_discovery/apps/api/tests/test_serializers.py]: It seems your find-replace operation did too much.

**Review by clintonb — CHANGES_REQUESTED** [review]

**Review by clintonb — COMMENTED** [review]
How will this data be used?

**Comment by MatthewPiatetsky** [comment]
We need the upgrade deadlines on seats for both the course about and program about pages to implement all the upgrade deadline logic for the multi course runs  
this data is already available in the courses endpoint, just not the program endpoint

File: course_discovery/apps/api/tests/test_serializers.py (hunk 2 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/tests/test_serializers.py]: are you asking why i changed seats to staff?
this test is testing the "use_full_course_serializer" argument by checking if seats are present or not, so if i add them to the minimal serializer i need a different field for this check

File: course_discovery/apps/api/serializers.py (hunk 1 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/serializers.py:386]: if we need this data for the program detail page we could use a different serializer, but then nothing would use this serializer. Minimal should just be the smallest set of an actual existing use case right?

File: course_discovery/apps/api/tests/test_serializers.py (hunk 2 above)
Comment (clintonb) [reply course_discovery/apps/api/tests/test_serializers.py]: Yeah...this test is not implemented well. If you want to verify the `CourseSerializer` is used, just use that class directly. Check to see what data *isn't* in the serialized value is odd and, as we see now, brittle. `test_course_ordering` offers a good example of cleanly validating the courses.

```python
actual = self.serializer_class(program, context={'request': request, 'use_full_course_serializer': 1}).data['courses']
expected = ProgramCourseSerializer(
    program.courses,
    many=True,
    context={'request': request, 'program': program, 'course_runs': list(program.course_runs)}
).data
self.assertDictEqual(actual, expected)
```

**Review by clintonb — CHANGES_REQUESTED** [review]

File: course_discovery/apps/api/tests/test_serializers.py (hunk 2 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/tests/test_serializers.py]: ok I changed the test to use the CourseRunSerializer

**Review by clintonb — APPROVED** [review]

**Pull request merged**